WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -40,6 +39,13 @@ private async Task<InputRegistrationResponse> RegisterInputCoreAsync(InputRegist
 		{
 			var round = GetRound(request.RoundId);
 
+			if (!OwnershipProof.VerifyCoinJoinInputProof(request.OwnershipProof, coin.ScriptPubKey, round.CoinJoinInputCommitmentData))
```

Comment:
Why not move line 70

_ = round.Assert<ConstructionState>().AddInput(coin, request.OwnershipProof, round.CoinJoinInputCommitmentData);

here?

Reply:
It could be, yes. Will play with this.

Reply:
Yes. Done. It is much better in this way.